src/ott/solvers/linear/univariate.py

```diff
@@ -11,336 +11,311 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-from typing import NamedTuple, Optional, Tuple, Union
+import functools
+from typing import NamedTuple, Optional, Tuple
 
 import jax
+import jax.experimental.sparse as jesp
 import jax.numpy as jnp
 
-from ott import utils
-from ott.geometry import costs, pointcloud
 from ott.math import utils as mu
 from ott.problems.linear import linear_problem
 
 __all__ = [
-    "UnivariateOutput", "UnivariateSolver", "uniform_distance",
-    "quantile_distance"
+    "UnivariateOutput",
+    "uniform_distance",
+    "quantile_distance",
```

Comment:
not comfortable with this name of `quantile_distance`, this was used as the name for an inner routine, originating in the original `scipy` implementation, which was only written for the 1-Wasserstein distance. But this has been adapted to fit any `TICost`, therefore not necessarily a distance.

Also, in principle, `north_west_distance` should output the same thing (except for dual vectors) as `quantile_distance`, so this is a bit confusing. Also, shouldn't be called a `distance`, these are fundamentally solvers that return optimal cost / OT primal variable  and possibly (for NW solver) the OT dual variable, not distances

Reply:
I also think, finally that a discussion should be introduced on which implementation to use. I expect the current `quantile_distance` to be faster than `north_west_distance`, hence I would only run the latter if `dual_variables` output is needed, which I don't expect to happen often.

Reply:
Done, thanks for the comment!